tests/convert_utf32_to_utf8_tests.cpp

```diff
@@ -150,4 +150,26 @@ TEST(special_cases) {
   ASSERT_EQUAL(utf8size, utf8len);
 }
 
+TEST(fuzz_7cfce3ad85893b9ca6431be21591ba8b1aa93713) {
+  const uint8_t input[] =
+      "\x13`"
+      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80\x00\x00\x00\x00\x00\x00\x00\x80"
+      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+      "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+      "\x00'"
+      "\x00\x00\x00\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+      "\xff\x00\x00\x00";
+
+  const char32_t *utf32 = reinterpret_cast<const char32_t *>(input);
```

Comment:
I think this is only legal as of C++20 (since this paper: https://wg21.link/P0593 ) and this is supposed to compile as C++11 as well.

to avoid this language lawyer gymnastics (which I am not very good at :) it would be easier to provide the data as char32_t directly?